--- a/config/DIABLO_SPEC_CONFIG_NAME.cfg	2008-03-06 01:16:50.000000000 +0100
+++ b/config/DIABLO_SPEC_CONFIG_NAME.cfg	2013-12-23 14:01:19.000000000 +0100
@@ -31,7 +31,7 @@
 
 ignore_errors = yes
 tune          = base
-ext           = i386-m32-gcc42-nn
+ext           = DIABLO_SPEC_CONFIG_NAME-nn
 output_format = asc, pdf, Screen, html
 reportable    = 1
 teeout        = yes
@@ -52,9 +52,11 @@
 # $LD_LIBRARY_PATH with something like
 # export LD_LIBRARY_PATH=/usr/local/gcc42-0715-32/lib
 #
-CC           = /usr/local/bin/gcc
-CXX          = /usr/local/bin/g++
-FC           = /usr/local/bin/gfortran
+CC           = DIABLO_CROSSTOOLS_INSTALLED_DIR/DIABLO_CROSSTOOLS_PREFIX/bin/DIABLO_CROSSTOOLS_PREFIX-gcc
+CXX          = DIABLO_CROSSTOOLS_INSTALLED_DIR/DIABLO_CROSSTOOLS_PREFIX/bin/DIABLO_CROSSTOOLS_PREFIX-g++
+FC           = DIABLO_CROSSTOOLS_INSTALLED_DIR/DIABLO_CROSSTOOLS_PREFIX/bin/DIABLO_CROSSTOOLS_PREFIX-gfortran
+
+EXTRA_LDFLAGS=-static -Wl,-Map,$@.map
 
 ## HW config
 hw_model     = Tyan Thunder KKQS Pro (S4882)
@@ -85,20 +87,20 @@
 sw_base_ptrsize = 32-bit
 sw_peak_ptrsize = 32-bit
 
-
+makeflags = -j SPEC_PARALLEL_BUILD_FACTOR
 #####################################################################
 # Optimization
 #####################################################################
 ## Base is low opt
 default=base=default=default:
 
-COPTIMIZE     = -O2
-CXXOPTIMIZE  = -O2 
-FOPTIMIZE    = -O2
-
-notes0100= C base flags: -O2
-notes0110= C++ base flags: -O2
-notes0120= Fortran base flags: -O2
+COPTIMIZE     = DIABLO_SPEC_OPTIMIZE_FLAGS
+CXXOPTIMIZE  = DIABLO_SPEC_OPTIMIZE_FLAGS
+FOPTIMIZE    = DIABLO_SPEC_OPTIMIZE_FLAGS
+
+notes0100= C base flags: DIABLO_SPEC_OPTIMIZE_FLAGS
+notes0110= C++ base flags: DIABLO_SPEC_OPTIMIZE_FLAGS
+notes0120= Fortran base flags: DIABLO_SPEC_OPTIMIZE_FLAGS
 
 #####################################################################
 # 32/64 bit Portability Flags - all
@@ -112,15 +114,21 @@
 #####################################################################
 
 400.perlbench=default=default=default:
-notes35    = 400.perlbench: -DSPEC_CPU_LINUX_IA32
-CPORTABILITY= -DSPEC_CPU_LINUX_IA32
+notes35    = 400.perlbench: -DSPEC_CPU_LINUX -std=gnu89
+CPORTABILITY= -DSPEC_CPU_LINUX -std=gnu89
 
 462.libquantum=default=default=default:
 notes60= 462.libquantum: -DSPEC_CPU_LINUX
 CPORTABILITY= -DSPEC_CPU_LINUX
 
+447.dealII=default=default=default:
+CXXPORTABILITY= -include cstddef -include cstring -include cstdlib
+
 483.xalancbmk=default=default=default:
-CXXPORTABILITY= -DSPEC_CPU_LINUX
+CXXPORTABILITY= -DSPEC_CPU_LINUX -include cstdlib -include cstring
+
+482.sphinx3=default=default=default:
+CPORTABILITY= -fsigned-char
 
 
 #####################################################################
