--- mibench/security/rijndael/aesxam.c	2001-10-17 05:12:15.000000000 +0200
+++ mibench/security/rijndael/aesxam.c	2012-06-21 10:55:43.000000000 +0200
@@ -38,7 +38,6 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <memory.h>
 #include <ctype.h>
 
 #include "aes.h"
@@ -97,12 +96,12 @@
 
 int encfile(FILE *fin, FILE *fout, aes *ctx, char* fn)
 {   char            inbuf[16], outbuf[16];
-    fpos_t          flen;
+    long            flen;
     unsigned long   i=0, l=0;
 
     fillrand(outbuf, 16);           /* set an IV for CBC mode           */
     fseek(fin, 0, SEEK_END);        /* get the length of the file       */
-    fgetpos(fin, &flen);            /* and then reset to start          */
+    flen = ftell(fin);              /* and then reset to start          */
     fseek(fin, 0, SEEK_SET);        
     fwrite(outbuf, 1, 16, fout);    /* write the IV to the output       */
     fillrand(inbuf, 1);             /* make top 4 bits of a byte random */
